Dockerfile

```diff
@@ -12,6 +12,8 @@ RUN apk add --no-cache \
   git \
   curl
 
+ARG GOPROXY
```

Comment:
GOPROXY argument is declared but never used. That should become an environment variable, like in line 23

Reply:
and that's what is expected. It becomes an ENV for the duration of the build. Doing so we can define GOPROXY when required from environment, but keep is unset to get default go get behaviour

sample Dockerfile:
```
FROM alpine
ARG foo
RUN env
```
```
$ docker build .
Step 3/3 : RUN env
 ---> Running in a405f4f8b109
HOSTNAME=a405f4f8b109
SHLVL=1
HOME=/root
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/
```

```
$ docker build  --build-arg foo .
Step 3/3 : RUN env
 ---> Running in a405f4f8b109
HOSTNAME=a405f4f8b109
SHLVL=1
HOME=/root
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/
```

```
$ foo=bar docker build  --build-arg foo .
Step 3/3 : RUN env
 ---> Running in a405f4f8b109
HOSTNAME=a405f4f8b109
foo=bar
SHLVL=1
HOME=/root
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/
```

Reply:
I'm not sure what line 23 is for, as those are already declared in env during the build, probably for tracking purpose but then I'd probably used a label.